okapi-core/src/test/java/org/folio/okapi/MainDeployTest.java

```diff
@@ -236,87 +189,53 @@ public void testClusterMode(TestContext context) {
         .then().statusCode(200).log().ifValidationFails().extract().response();
       Assert.assertTrue("raml: " + c.getLastReport().toString(),
         c.getLastReport().isEmpty());
-      async.complete();
-    });
+      vertx.close(context.asyncAssertSuccess(x -> async.complete()));
+    }));
+    async.await();
   }
 
   @Test
   public void testClusterModeFail1(TestContext context) {
-    async = context.async();
-
     String[] args = {"cluster", "-cluster-host", "foobar", "-cluster-port", "5701"};
 
     MainDeploy d = new MainDeploy();
-    d.init(args, res -> {
-      vertx = res.succeeded() ? res.result() : null;
-      Assert.assertTrue("main1 " + res.cause(), res.failed());
-      async.complete();
-    });
-    async.await(1000);
+    d.init(args, context.asyncAssertFailure());
   }
 
   @Test
   public void testClusterModeFail2(TestContext context) {
-    async = context.async();
-
     String[] args = {"cluster", "-hazelcast-config-file", "foobar"};
-
     MainDeploy d = new MainDeploy();
-    d.init(args, res -> {
-      vertx = res.succeeded() ? res.result() : null;
-      Assert.assertTrue("main1 " + res.cause(), res.failed());
-      async.complete();
-    });
-    async.await(1000);
+    d.init(args, context.asyncAssertFailure());
   }
 
   @Test
   public void testClusterModeFail3(TestContext context) {
-    async = context.async();
-
     String[] args = {"cluster", "-hazelcast-config-cp", "foobar"};
-
     MainDeploy d = new MainDeploy();
-    d.init(args, res -> {
-      vertx = res.succeeded() ? res.result() : null;
-      Assert.assertTrue("main1 " + res.cause(), res.failed());
-      async.complete();
-    });
-    async.await(1000);
+    d.init(args, context.asyncAssertFailure());
   }
 
   @Test
   public void testClusterModeFail4(TestContext context) {
-    async = context.async();
-
     String[] args = {"cluster", "-hazelcast-config-url", "foobar"};
 
     MainDeploy d = new MainDeploy();
-    d.init(args, res -> {
-      vertx = res.succeeded() ? res.result() : null;
-      Assert.assertTrue("main1 " + res.cause(), res.failed());
-      async.complete();
-    });
-    async.await(1000);
+    d.init(args, context.asyncAssertFailure());
   }
 
   @Test
   public void testOkapiSamePort(TestContext context) {
-    async = context.async();
-
     String[] args = {"dev"};
 
+    Async async = context.async();
     MainDeploy d1 = new MainDeploy();
-    d1.init(args, res1 -> {
-      vertx = res1.succeeded() ? res1.result() : null;
-      Assert.assertTrue("d1 " + res1.cause(), res1.succeeded());
-
+    d1.init(args, context.asyncAssertSuccess(vertx -> {
       MainDeploy d2 = new MainDeploy();
-      d2.init(args, res2 -> {
-        Assert.assertTrue("d2 " + res2.cause(), res2.failed());
-        async.complete();
-      });
-    });
+      d2.init(args, context.asyncAssertFailure(
+          x -> vertx.close(context.asyncAssertSuccess(y -> async.countDown()))));
```

Comment:
countDown() should be replaced by complete()

Reply:
Jesus!. Yes.